In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [3]:
df = pd.read_csv("houseprice.csv")

In [5]:
df

Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0        1          60       RL         65.0     8450   Pave   NaN      Reg   
1        2          20       RL         80.0     9600   Pave   NaN      Reg   
2        3          60       RL         68.0    11250   Pave   NaN      IR1   
3        4          70       RL         60.0     9550   Pave   NaN      IR1   
4        5          60       RL         84.0    14260   Pave   NaN      IR1   
...    ...         ...      ...          ...      ...    ...   ...      ...   
1455  1456          60       RL         62.0     7917   Pave   NaN      Reg   
1456  1457          20       RL         85.0    13175   Pave   NaN      Reg   
1457  1458          70       RL         66.0     9042   Pave   NaN      Reg   
1458  1459          20       RL         68.0     9717   Pave   NaN      Reg   
1459  1460          20       RL         75.0     9937   Pave   NaN      Reg   

     LandContour Utilities  ... PoolArea PoolQC  Fence MiscFeature MiscVal  \
0            Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
1            Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
2            Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
3            Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
4            Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
...          ...       ...  ...      ...    ...    ...         ...     ...   
1455         Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
1456         Lvl    AllPub  ...        0    NaN  MnPrv         NaN       0   
1457         Lvl    AllPub  ...        0    NaN  GdPrv        Shed    2500   
1458         Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
1459         Lvl    AllPub  ...        0    NaN    NaN         NaN       0   

     MoSold YrSold  SaleType  SaleCondition  SalePrice  
0         2   2008        WD         Normal     208500  
1         5   2007        WD         Normal     181500  
2         9   2008        WD         Normal     223500  
3         2   2006        WD        Abnorml     140000  
4        12   2008        WD         Normal     250000  
...     ...    ...       ...            ...        ...  
1455      8   2007        WD         Normal     175000  
1456      2   2010        WD         Normal     210000  
1457      5   2010        WD         Normal     266500  
1458      4   2010        WD         Normal     142125  
1459      6   2008        WD         Normal     147500  

[1460 rows x 81 columns]

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 81 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             1460 non-null   int64  
 1   MSSubClass     1460 non-null   int64  
 2   MSZoning       1460 non-null   object 
 3   LotFrontage    1201 non-null   float64
 4   LotArea        1460 non-null   int64  
 5   Street         1460 non-null   object 
 6   Alley          91 non-null     object 
 7   LotShape       1460 non-null   object 
 8   LandContour    1460 non-null   object 
 9   Utilities      1460 non-null   object 
 10  LotConfig      1460 non-null   object 
 11  LandSlope      1460 non-null   object 
 12  Neighborhood   1460 non-null   object 
 13  Condition1     1460 non-null   object 
 14  Condition2     1460 non-null   object 
 15  BldgType       1460 non-null   object 
 16  HouseStyle     1460 non-null   object 
 17  OverallQual    1460 non-null   int64  
 18  OverallC

In [20]:
df.columns

Index(['Id', 'MSSubClass', 'MSZoning', 'LotFrontage', 'LotArea', 'Street',
       'Alley', 'LotShape', 'LandContour', 'Utilities', 'LotConfig',
       'LandSlope', 'Neighborhood', 'Condition1', 'Condition2', 'BldgType',
       'HouseStyle', 'OverallQual', 'OverallCond', 'YearBuilt', 'YearRemodAdd',
       'RoofStyle', 'RoofMatl', 'Exterior1st', 'Exterior2nd', 'MasVnrType',
       'MasVnrArea', 'ExterQual', 'ExterCond', 'Foundation', 'BsmtQual',
       'BsmtCond', 'BsmtExposure', 'BsmtFinType1', 'BsmtFinSF1',
       'BsmtFinType2', 'BsmtFinSF2', 'BsmtUnfSF', 'TotalBsmtSF', 'Heating',
       'HeatingQC', 'CentralAir', 'Electrical', '1stFlrSF', '2ndFlrSF',
       'LowQualFinSF', 'GrLivArea', 'BsmtFullBath', 'BsmtHalfBath', 'FullBath',
       'HalfBath', 'BedroomAbvGr', 'KitchenAbvGr', 'KitchenQual',
       'TotRmsAbvGrd', 'Functional', 'Fireplaces', 'FireplaceQu', 'GarageType',
       'GarageYrBlt', 'GarageFinish', 'GarageCars', 'GarageArea', 'GarageQual',
       'GarageCond', 'PavedDrive

In [21]:
(df.isnull().sum() * 100 / df.shape[0]).head(40)

Id               0.000000
MSSubClass       0.000000
MSZoning         0.000000
LotFrontage     17.739726
LotArea          0.000000
Street           0.000000
Alley           93.767123
LotShape         0.000000
LandContour      0.000000
Utilities        0.000000
LotConfig        0.000000
LandSlope        0.000000
Neighborhood     0.000000
Condition1       0.000000
Condition2       0.000000
BldgType         0.000000
HouseStyle       0.000000
OverallQual      0.000000
OverallCond      0.000000
YearBuilt        0.000000
YearRemodAdd     0.000000
RoofStyle        0.000000
RoofMatl         0.000000
Exterior1st      0.000000
Exterior2nd      0.000000
MasVnrType       0.547945
MasVnrArea       0.547945
ExterQual        0.000000
ExterCond        0.000000
Foundation       0.000000
BsmtQual         2.534247
BsmtCond         2.534247
BsmtExposure     2.602740
BsmtFinType1     2.534247
BsmtFinSF1       0.000000
BsmtFinType2     2.602740
BsmtFinSF2       0.000000
BsmtUnfSF        0.000000
TotalBsmtSF 

In [22]:
(df.isnull().sum() * 100 / df.shape[0]).tail(40)

CentralAir        0.000000
Electrical        0.068493
1stFlrSF          0.000000
2ndFlrSF          0.000000
LowQualFinSF      0.000000
GrLivArea         0.000000
BsmtFullBath      0.000000
BsmtHalfBath      0.000000
FullBath          0.000000
HalfBath          0.000000
BedroomAbvGr      0.000000
KitchenAbvGr      0.000000
KitchenQual       0.000000
TotRmsAbvGrd      0.000000
Functional        0.000000
Fireplaces        0.000000
FireplaceQu      47.260274
GarageType        5.547945
GarageYrBlt       5.547945
GarageFinish      5.547945
GarageCars        0.000000
GarageArea        0.000000
GarageQual        5.547945
GarageCond        5.547945
PavedDrive        0.000000
WoodDeckSF        0.000000
OpenPorchSF       0.000000
EnclosedPorch     0.000000
3SsnPorch         0.000000
ScreenPorch       0.000000
PoolArea          0.000000
PoolQC           99.520548
Fence            80.753425
MiscFeature      96.301370
MiscVal           0.000000
MoSold            0.000000
YrSold            0.000000
S

In [33]:
notenough_frag = (df.isnull().sum() * 100 / df.shape[0]) > 40

notenough_info = df.columns[notenough_frag]

notenough_info

Index(['Alley', 'FireplaceQu', 'PoolQC', 'Fence', 'MiscFeature'], dtype='object')

In [34]:
for column in notenough_info:
    del df[column]

In [55]:
missing_frag = df.isnull().sum() * 100 / df.shape[0] > 0

missing_info = df.columns[missing_frag]

df[missing_info]

LotFrontage MasVnrType  MasVnrArea BsmtQual BsmtCond BsmtExposure  \
0            65.0    BrkFace       196.0       Gd       TA           No   
1            80.0       None         0.0       Gd       TA           Gd   
2            68.0    BrkFace       162.0       Gd       TA           Mn   
3            60.0       None         0.0       TA       Gd           No   
4            84.0    BrkFace       350.0       Gd       TA           Av   
...           ...        ...         ...      ...      ...          ...   
1455         62.0       None         0.0       Gd       TA           No   
1456         85.0      Stone       119.0       Gd       TA           No   
1457         66.0       None         0.0       TA       Gd           No   
1458         68.0       None         0.0       TA       TA           Mn   
1459         75.0       None         0.0       TA       TA           No   

     BsmtFinType1 BsmtFinType2 Electrical GarageType  GarageYrBlt  \
0             GLQ          Unf      SBrkr     Attchd       2003.0   
1             ALQ          Unf      SBrkr     Attchd       1976.0   
2             GLQ          Unf      SBrkr     Attchd       2001.0   
3             ALQ          Unf      SBrkr     Detchd       1998.0   
4             GLQ          Unf      SBrkr     Attchd       2000.0   
...           ...          ...        ...        ...          ...   
1455          Unf          Unf      SBrkr     Attchd       1999.0   
1456          ALQ          Rec      SBrkr     Attchd       1978.0   
1457          GLQ          Unf      SBrkr     Attchd       1941.0   
1458          GLQ          Rec      FuseA     Attchd       1950.0   
1459          BLQ          LwQ      SBrkr     Attchd       1965.0   

     GarageFinish GarageQual GarageCond  
0             RFn         TA         TA  
1             RFn         TA         TA  
2             RFn         TA         TA  
3             Unf         TA         TA  
4             RFn         TA         TA  
...           ...        ...        ...  
1455          RFn         TA         TA  
1456          Unf         TA         TA  
1457          RFn         TA         TA  
1458          Unf         TA         TA  
1459          Fin         TA         TA  

[1460 rows x 14 columns]

I think only "Electrical" column is important above these data, so i only need to fill "Electrical" column 

In [58]:
df["Electrical"].isnull().sum()

1

In [86]:
df.loc[df.Electrical[df["Electrical"].isnull()].index]

Empty DataFrame
Columns: [Id, MSSubClass, MSZoning, LotFrontage, LotArea, Street, LotShape, LandContour, Utilities, LotConfig, LandSlope, Neighborhood, Condition1, Condition2, BldgType, HouseStyle, OverallQual, OverallCond, YearBuilt, YearRemodAdd, RoofStyle, RoofMatl, Exterior1st, Exterior2nd, MasVnrType, MasVnrArea, ExterQual, ExterCond, Foundation, BsmtQual, BsmtCond, BsmtExposure, BsmtFinType1, BsmtFinSF1, BsmtFinType2, BsmtFinSF2, BsmtUnfSF, TotalBsmtSF, Heating, HeatingQC, CentralAir, Electrical, 1stFlrSF, 2ndFlrSF, LowQualFinSF, GrLivArea, BsmtFullBath, BsmtHalfBath, FullBath, HalfBath, BedroomAbvGr, KitchenAbvGr, KitchenQual, TotRmsAbvGrd, Functional, Fireplaces, GarageType, GarageYrBlt, GarageFinish, GarageCars, GarageArea, GarageQual, GarageCond, PavedDrive, WoodDeckSF, OpenPorchSF, EnclosedPorch, 3SsnPorch, ScreenPorch, PoolArea, MiscVal, MoSold, YrSold, SaleType, SaleCondition, SalePrice]
Index: []

[0 rows x 76 columns]

In [85]:
df.groupby(by="Electrical").count()

Id  MSSubClass  MSZoning  LotFrontage  LotArea  Street  \
Electrical                                                             
FuseA         95          95        95           81       95      95   
FuseF         27          27        27           25       27      27   
FuseP          3           3         3            3        3       3   
Mix            1           1         1            1        1       1   
SBrkr       1334        1334      1334         1091     1334    1334   

            LotShape  LandContour  Utilities  LotConfig  ...  EnclosedPorch  \
Electrical                                               ...                  
FuseA             95           95         95         95  ...             95   
FuseF             27           27         27         27  ...             27   
FuseP              3            3          3          3  ...              3   
Mix                1            1          1          1  ...              1   
SBrkr           1334         1334       1334       1334  ...           1334   

            3SsnPorch  ScreenPorch  PoolArea  MiscVal  MoSold  YrSold  \
Electrical                                                              
FuseA              95           95        95       95      95      95   
FuseF              27           27        27       27      27      27   
FuseP               3            3         3        3       3       3   
Mix                 1            1         1        1       1       1   
SBrkr            1334         1334      1334     1334    1334    1334   

            SaleType  SaleCondition  SalePrice  
Electrical                                      
FuseA             95             95         95  
FuseF             27             27         27  
FuseP              3              3          3  
Mix                1              1          1  
SBrkr           1334           1334       1334  

[5 rows x 75 columns]

In [83]:
df.loc[1379,"Electrical"] = "FuseA"

In [107]:
missing_frag = df.isnull().sum() * 100 / df.shape[0] > 0

missing_info = df.columns[missing_frag]

df[missing_info]

LotFrontage MasVnrType  MasVnrArea BsmtQual BsmtCond BsmtExposure  \
0            65.0    BrkFace       196.0       Gd       TA           No   
1            80.0       None         0.0       Gd       TA           Gd   
2            68.0    BrkFace       162.0       Gd       TA           Mn   
3            60.0       None         0.0       TA       Gd           No   
4            84.0    BrkFace       350.0       Gd       TA           Av   
...           ...        ...         ...      ...      ...          ...   
1455         62.0       None         0.0       Gd       TA           No   
1456         85.0      Stone       119.0       Gd       TA           No   
1457         66.0       None         0.0       TA       Gd           No   
1458         68.0       None         0.0       TA       TA           Mn   
1459         75.0       None         0.0       TA       TA           No   

     BsmtFinType1 BsmtFinType2 GarageType  GarageYrBlt GarageFinish  \
0             GLQ          Unf     Attchd       2003.0          RFn   
1             ALQ          Unf     Attchd       1976.0          RFn   
2             GLQ          Unf     Attchd       2001.0          RFn   
3             ALQ          Unf     Detchd       1998.0          Unf   
4             GLQ          Unf     Attchd       2000.0          RFn   
...           ...          ...        ...          ...          ...   
1455          Unf          Unf     Attchd       1999.0          RFn   
1456          ALQ          Rec     Attchd       1978.0          Unf   
1457          GLQ          Unf     Attchd       1941.0          RFn   
1458          GLQ          Rec     Attchd       1950.0          Unf   
1459          BLQ          LwQ     Attchd       1965.0          Fin   

     GarageQual GarageCond  
0            TA         TA  
1            TA         TA  
2            TA         TA  
3            TA         TA  
4            TA         TA  
...         ...        ...  
1455         TA         TA  
1456         TA         TA  
1457         TA         TA  
1458         TA         TA  
1459         TA         TA  

[1460 rows x 13 columns]

In [91]:
df.loc[df.Electrical[df["GarageType"].isnull()].index]

Id  MSSubClass MSZoning  LotFrontage  LotArea Street LotShape  \
39      40          90       RL         65.0     6040   Pave      Reg   
48      49         190       RM         33.0     4456   Pave      Reg   
78      79          90       RL         72.0    10778   Pave      Reg   
88      89          50  C (all)        105.0     8470   Pave      IR1   
89      90          20       RL         60.0     8070   Pave      Reg   
...    ...         ...      ...          ...      ...    ...      ...   
1349  1350          70       RM         50.0     5250   Pave      Reg   
1407  1408          20       RL          NaN     8780   Pave      IR1   
1449  1450         180       RM         21.0     1533   Pave      Reg   
1450  1451          90       RL         60.0     9000   Pave      Reg   
1453  1454          20       RL         90.0    17217   Pave      Reg   

     LandContour Utilities LotConfig  ... EnclosedPorch 3SsnPorch ScreenPorch  \
39           Lvl    AllPub    Inside  ...             0         0           0   
48           Lvl    AllPub    Inside  ...           102         0           0   
78           Lvl    AllPub    Inside  ...             0         0           0   
88           Lvl    AllPub    Corner  ...           156         0           0   
89           Lvl    AllPub    Inside  ...             0         0           0   
...          ...       ...       ...  ...           ...       ...         ...   
1349         Lvl    AllPub    Inside  ...            20         0           0   
1407         Lvl    AllPub    Corner  ...             0         0           0   
1449         Lvl    AllPub    Inside  ...             0         0           0   
1450         Lvl    AllPub       FR2  ...             0         0           0   
1453         Lvl    AllPub    Inside  ...             0         0           0   

     PoolArea MiscVal MoSold  YrSold  SaleType  SaleCondition  SalePrice  
39          0       0      6    2008        WD        AdjLand      82000  
48          0       0      6    2009       New        Partial     113000  
78          0       0      4    2010        WD         Normal     136500  
88          0       0     10    2009     ConLD        Abnorml      85000  
89          0       0      8    2007        WD         Normal     123600  
...       ...     ...    ...     ...       ...            ...        ...  
1349        0       0     12    2008        WD         Normal     122000  
1407        0       0      3    2009        WD         Normal     112000  
1449        0       0      8    2006        WD        Abnorml      92000  
1450        0       0      9    2009        WD         Normal     136000  
1453        0       0      7    2006        WD        Abnorml      84500  

[81 rows x 76 columns]

In [92]:
df.loc[df.Electrical[df["GarageQual"].isnull()].index]

Id  MSSubClass MSZoning  LotFrontage  LotArea Street LotShape  \
39      40          90       RL         65.0     6040   Pave      Reg   
48      49         190       RM         33.0     4456   Pave      Reg   
78      79          90       RL         72.0    10778   Pave      Reg   
88      89          50  C (all)        105.0     8470   Pave      IR1   
89      90          20       RL         60.0     8070   Pave      Reg   
...    ...         ...      ...          ...      ...    ...      ...   
1349  1350          70       RM         50.0     5250   Pave      Reg   
1407  1408          20       RL          NaN     8780   Pave      IR1   
1449  1450         180       RM         21.0     1533   Pave      Reg   
1450  1451          90       RL         60.0     9000   Pave      Reg   
1453  1454          20       RL         90.0    17217   Pave      Reg   

     LandContour Utilities LotConfig  ... EnclosedPorch 3SsnPorch ScreenPorch  \
39           Lvl    AllPub    Inside  ...             0         0           0   
48           Lvl    AllPub    Inside  ...           102         0           0   
78           Lvl    AllPub    Inside  ...             0         0           0   
88           Lvl    AllPub    Corner  ...           156         0           0   
89           Lvl    AllPub    Inside  ...             0         0           0   
...          ...       ...       ...  ...           ...       ...         ...   
1349         Lvl    AllPub    Inside  ...            20         0           0   
1407         Lvl    AllPub    Corner  ...             0         0           0   
1449         Lvl    AllPub    Inside  ...             0         0           0   
1450         Lvl    AllPub       FR2  ...             0         0           0   
1453         Lvl    AllPub    Inside  ...             0         0           0   

     PoolArea MiscVal MoSold  YrSold  SaleType  SaleCondition  SalePrice  
39          0       0      6    2008        WD        AdjLand      82000  
48          0       0      6    2009       New        Partial     113000  
78          0       0      4    2010        WD         Normal     136500  
88          0       0     10    2009     ConLD        Abnorml      85000  
89          0       0      8    2007        WD         Normal     123600  
...       ...     ...    ...     ...       ...            ...        ...  
1349        0       0     12    2008        WD         Normal     122000  
1407        0       0      3    2009        WD         Normal     112000  
1449        0       0      8    2006        WD        Abnorml      92000  
1450        0       0      9    2009        WD         Normal     136000  
1453        0       0      7    2006        WD        Abnorml      84500  

[81 rows x 76 columns]

In [104]:
df.loc[df.Electrical[df["GarageCond"].isnull()].index]

Id  MSSubClass MSZoning  LotFrontage  LotArea Street LotShape  \
39      40          90       RL         65.0     6040   Pave      Reg   
48      49         190       RM         33.0     4456   Pave      Reg   
78      79          90       RL         72.0    10778   Pave      Reg   
88      89          50  C (all)        105.0     8470   Pave      IR1   
89      90          20       RL         60.0     8070   Pave      Reg   
...    ...         ...      ...          ...      ...    ...      ...   
1349  1350          70       RM         50.0     5250   Pave      Reg   
1407  1408          20       RL          NaN     8780   Pave      IR1   
1449  1450         180       RM         21.0     1533   Pave      Reg   
1450  1451          90       RL         60.0     9000   Pave      Reg   
1453  1454          20       RL         90.0    17217   Pave      Reg   

     LandContour Utilities LotConfig  ... EnclosedPorch 3SsnPorch ScreenPorch  \
39           Lvl    AllPub    Inside  ...             0         0           0   
48           Lvl    AllPub    Inside  ...           102         0           0   
78           Lvl    AllPub    Inside  ...             0         0           0   
88           Lvl    AllPub    Corner  ...           156         0           0   
89           Lvl    AllPub    Inside  ...             0         0           0   
...          ...       ...       ...  ...           ...       ...         ...   
1349         Lvl    AllPub    Inside  ...            20         0           0   
1407         Lvl    AllPub    Corner  ...             0         0           0   
1449         Lvl    AllPub    Inside  ...             0         0           0   
1450         Lvl    AllPub       FR2  ...             0         0           0   
1453         Lvl    AllPub    Inside  ...             0         0           0   

     PoolArea MiscVal MoSold  YrSold  SaleType  SaleCondition  SalePrice  
39          0       0      6    2008        WD        AdjLand      82000  
48          0       0      6    2009       New        Partial     113000  
78          0       0      4    2010        WD         Normal     136500  
88          0       0     10    2009     ConLD        Abnorml      85000  
89          0       0      8    2007        WD         Normal     123600  
...       ...     ...    ...     ...       ...            ...        ...  
1349        0       0     12    2008        WD         Normal     122000  
1407        0       0      3    2009        WD         Normal     112000  
1449        0       0      8    2006        WD        Abnorml      92000  
1450        0       0      9    2009        WD         Normal     136000  
1453        0       0      7    2006        WD        Abnorml      84500  

[81 rows x 76 columns]

In [113]:
df = df.drop(df.Electrical[df["GarageCond"].isnull()].index)

In [114]:
missing_frag = df.isnull().sum() * 100 / df.shape[0] > 0

missing_info = df.columns[missing_frag]

df[missing_info]

LotFrontage MasVnrType  MasVnrArea BsmtQual BsmtCond BsmtExposure  \
0            65.0    BrkFace       196.0       Gd       TA           No   
1            80.0       None         0.0       Gd       TA           Gd   
2            68.0    BrkFace       162.0       Gd       TA           Mn   
3            60.0       None         0.0       TA       Gd           No   
4            84.0    BrkFace       350.0       Gd       TA           Av   
...           ...        ...         ...      ...      ...          ...   
1455         62.0       None         0.0       Gd       TA           No   
1456         85.0      Stone       119.0       Gd       TA           No   
1457         66.0       None         0.0       TA       Gd           No   
1458         68.0       None         0.0       TA       TA           Mn   
1459         75.0       None         0.0       TA       TA           No   

     BsmtFinType1 BsmtFinType2  
0             GLQ          Unf  
1             ALQ          Unf  
2             GLQ          Unf  
3             ALQ          Unf  
4             GLQ          Unf  
...           ...          ...  
1455          Unf          Unf  
1456          ALQ          Rec  
1457          GLQ          Unf  
1458          GLQ          Rec  
1459          BLQ          LwQ  

[1379 rows x 8 columns]

In [122]:
for column in df[missing_info].columns:
    del df[column]
    
df

KeyError: "None of [Index(['LotFrontage', 'MasVnrType', 'MasVnrArea', 'BsmtQual', 'BsmtCond',\n       'BsmtExposure', 'BsmtFinType1', 'BsmtFinType2'],\n      dtype='object')] are in the [columns]"

In [124]:
missing_frag = df.isnull().sum() * 100 / df.shape[0] > 0

missing_info = df.columns[missing_frag]

df[missing_info]

Empty DataFrame
Columns: []
Index: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 40, 41, 42, 43, 44, 45, 46, 47, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 79, 80, 81, 82, 83, 84, 85, 86, 87, 90, 91, 92, 93, 94, 95, 96, 97, 98, 100, 101, 102, 103, 104, 105, ...]

[1379 rows x 0 columns]

I want to see how this raw data will work with my model. After trying with these raw data I will manually add/drop some features and try again.

In [127]:
df.to_csv("houseprices_filled_raw.csv")

MSSubClass: Identifies the type of dwelling involved in the sale.	

        20	1-STORY 1946 & NEWER ALL STYLES
        30	1-STORY 1945 & OLDER
        40	1-STORY W/FINISHED ATTIC ALL AGES
        45	1-1/2 STORY - UNFINISHED ALL AGES
        50	1-1/2 STORY FINISHED ALL AGES
        60	2-STORY 1946 & NEWER
        70	2-STORY 1945 & OLDER
        75	2-1/2 STORY ALL AGES
        80	SPLIT OR MULTI-LEVEL
        85	SPLIT FOYER
        90	DUPLEX - ALL STYLES AND AGES
       120	1-STORY PUD (Planned Unit Development) - 1946 & NEWER
       150	1-1/2 STORY PUD - ALL AGES
       160	2-STORY PUD - 1946 & NEWER
       180	PUD - MULTILEVEL - INCL SPLIT LEV/FOYER
       190	2 FAMILY CONVERSION - ALL STYLES AND AGES

MSZoning: Identifies the general zoning classification of the sale.
		
       A	Agriculture
       C	Commercial
       FV	Floating Village Residential
       I	Industrial
       RH	Residential High Density
       RL	Residential Low Density
       RP	Residential Low Density Park 
       RM	Residential Medium Density
	
LotFrontage: Linear feet of street connected to property

LotArea: Lot size in square feet

Street: Type of road access to property

       Grvl	Gravel	
       Pave	Paved
       	
Alley: Type of alley access to property

       Grvl	Gravel
       Pave	Paved
       NA 	No alley access
		
LotShape: General shape of property

       Reg	Regular	
       IR1	Slightly irregular
       IR2	Moderately Irregular
       IR3	Irregular
       
LandContour: Flatness of the property

       Lvl	Near Flat/Level	
       Bnk	Banked - Quick and significant rise from street grade to building
       HLS	Hillside - Significant slope from side to side
       Low	Depression
		
Utilities: Type of utilities available
		
       AllPub	All public Utilities (E,G,W,& S)	
       NoSewr	Electricity, Gas, and Water (Septic Tank)
       NoSeWa	Electricity and Gas Only
       ELO	Electricity only	
	
LotConfig: Lot configuration

       Inside	Inside lot
       Corner	Corner lot
       CulDSac	Cul-de-sac
       FR2	Frontage on 2 sides of property
       FR3	Frontage on 3 sides of property
	
LandSlope: Slope of property
		
       Gtl	Gentle slope
       Mod	Moderate Slope	
       Sev	Severe Slope
	
Neighborhood: Physical locations within Ames city limits

       Blmngtn	Bloomington Heights
       Blueste	Bluestem
       BrDale	Briardale
       BrkSide	Brookside
       ClearCr	Clear Creek
       CollgCr	College Creek
       Crawfor	Crawford
       Edwards	Edwards
       Gilbert	Gilbert
       IDOTRR	Iowa DOT and Rail Road
       MeadowV	Meadow Village
       Mitchel	Mitchell
       Names	North Ames
       NoRidge	Northridge
       NPkVill	Northpark Villa
       NridgHt	Northridge Heights
       NWAmes	Northwest Ames
       OldTown	Old Town
       SWISU	South & West of Iowa State University
       Sawyer	Sawyer
       SawyerW	Sawyer West
       Somerst	Somerset
       StoneBr	Stone Brook
       Timber	Timberland
       Veenker	Veenker
			
Condition1: Proximity to various conditions
	
       Artery	Adjacent to arterial street
       Feedr	Adjacent to feeder street	
       Norm	Normal	
       RRNn	Within 200' of North-South Railroad
       RRAn	Adjacent to North-South Railroad
       PosN	Near positive off-site feature--park, greenbelt, etc.
       PosA	Adjacent to postive off-site feature
       RRNe	Within 200' of East-West Railroad
       RRAe	Adjacent to East-West Railroad
	
Condition2: Proximity to various conditions (if more than one is present)
		
       Artery	Adjacent to arterial street
       Feedr	Adjacent to feeder street	
       Norm	Normal	
       RRNn	Within 200' of North-South Railroad
       RRAn	Adjacent to North-South Railroad
       PosN	Near positive off-site feature--park, greenbelt, etc.
       PosA	Adjacent to postive off-site feature
       RRNe	Within 200' of East-West Railroad
       RRAe	Adjacent to East-West Railroad
	
BldgType: Type of dwelling
		
       1Fam	Single-family Detached	
       2FmCon	Two-family Conversion; originally built as one-family dwelling
       Duplx	Duplex
       TwnhsE	Townhouse End Unit
       TwnhsI	Townhouse Inside Unit
	
HouseStyle: Style of dwelling
	
       1Story	One story
       1.5Fin	One and one-half story: 2nd level finished
       1.5Unf	One and one-half story: 2nd level unfinished
       2Story	Two story
       2.5Fin	Two and one-half story: 2nd level finished
       2.5Unf	Two and one-half story: 2nd level unfinished
       SFoyer	Split Foyer
       SLvl	Split Level
	
OverallQual: Rates the overall material and finish of the house

       10	Very Excellent
       9	Excellent
       8	Very Good
       7	Good
       6	Above Average
       5	Average
       4	Below Average
       3	Fair
       2	Poor
       1	Very Poor
	
OverallCond: Rates the overall condition of the house

       10	Very Excellent
       9	Excellent
       8	Very Good
       7	Good
       6	Above Average	
       5	Average
       4	Below Average	
       3	Fair
       2	Poor
       1	Very Poor
		
YearBuilt: Original construction date

YearRemodAdd: Remodel date (same as construction date if no remodeling or additions)

RoofStyle: Type of roof

       Flat	Flat
       Gable	Gable
       Gambrel	Gabrel (Barn)
       Hip	Hip
       Mansard	Mansard
       Shed	Shed
		
RoofMatl: Roof material

       ClyTile	Clay or Tile
       CompShg	Standard (Composite) Shingle
       Membran	Membrane
       Metal	Metal
       Roll	Roll
       Tar&Grv	Gravel & Tar
       WdShake	Wood Shakes
       WdShngl	Wood Shingles
		
Exterior1st: Exterior covering on house

       AsbShng	Asbestos Shingles
       AsphShn	Asphalt Shingles
       BrkComm	Brick Common
       BrkFace	Brick Face
       CBlock	Cinder Block
       CemntBd	Cement Board
       HdBoard	Hard Board
       ImStucc	Imitation Stucco
       MetalSd	Metal Siding
       Other	Other
       Plywood	Plywood
       PreCast	PreCast	
       Stone	Stone
       Stucco	Stucco
       VinylSd	Vinyl Siding
       Wd Sdng	Wood Siding
       WdShing	Wood Shingles
	
Exterior2nd: Exterior covering on house (if more than one material)

       AsbShng	Asbestos Shingles
       AsphShn	Asphalt Shingles
       BrkComm	Brick Common
       BrkFace	Brick Face
       CBlock	Cinder Block
       CemntBd	Cement Board
       HdBoard	Hard Board
       ImStucc	Imitation Stucco
       MetalSd	Metal Siding
       Other	Other
       Plywood	Plywood
       PreCast	PreCast
       Stone	Stone
       Stucco	Stucco
       VinylSd	Vinyl Siding
       Wd Sdng	Wood Siding
       WdShing	Wood Shingles
	
MasVnrType: Masonry veneer type

       BrkCmn	Brick Common
       BrkFace	Brick Face
       CBlock	Cinder Block
       None	None
       Stone	Stone
	
MasVnrArea: Masonry veneer area in square feet

ExterQual: Evaluates the quality of the material on the exterior 
		
       Ex	Excellent
       Gd	Good
       TA	Average/Typical
       Fa	Fair
       Po	Poor
		
ExterCond: Evaluates the present condition of the material on the exterior
		
       Ex	Excellent
       Gd	Good
       TA	Average/Typical
       Fa	Fair
       Po	Poor
		
Foundation: Type of foundation
		
       BrkTil	Brick & Tile
       CBlock	Cinder Block
       PConc	Poured Contrete	
       Slab	Slab
       Stone	Stone
       Wood	Wood
		
BsmtQual: Evaluates the height of the basement

       Ex	Excellent (100+ inches)	
       Gd	Good (90-99 inches)
       TA	Typical (80-89 inches)
       Fa	Fair (70-79 inches)
       Po	Poor (<70 inches
       NA	No Basement
		
BsmtCond: Evaluates the general condition of the basement

       Ex	Excellent
       Gd	Good
       TA	Typical - slight dampness allowed
       Fa	Fair - dampness or some cracking or settling
       Po	Poor - Severe cracking, settling, or wetness
       NA	No Basement
	
BsmtExposure: Refers to walkout or garden level walls

       Gd	Good Exposure
       Av	Average Exposure (split levels or foyers typically score average or above)	
       Mn	Mimimum Exposure
       No	No Exposure
       NA	No Basement
	
BsmtFinType1: Rating of basement finished area

       GLQ	Good Living Quarters
       ALQ	Average Living Quarters
       BLQ	Below Average Living Quarters	
       Rec	Average Rec Room
       LwQ	Low Quality
       Unf	Unfinshed
       NA	No Basement
		
BsmtFinSF1: Type 1 finished square feet

BsmtFinType2: Rating of basement finished area (if multiple types)

       GLQ	Good Living Quarters
       ALQ	Average Living Quarters
       BLQ	Below Average Living Quarters	
       Rec	Average Rec Room
       LwQ	Low Quality
       Unf	Unfinshed
       NA	No Basement

BsmtFinSF2: Type 2 finished square feet

BsmtUnfSF: Unfinished square feet of basement area

TotalBsmtSF: Total square feet of basement area

Heating: Type of heating
		
       Floor	Floor Furnace
       GasA	Gas forced warm air furnace
       GasW	Gas hot water or steam heat
       Grav	Gravity furnace	
       OthW	Hot water or steam heat other than gas
       Wall	Wall furnace
		
HeatingQC: Heating quality and condition

       Ex	Excellent
       Gd	Good
       TA	Average/Typical
       Fa	Fair
       Po	Poor
		
CentralAir: Central air conditioning

       N	No
       Y	Yes
		
Electrical: Electrical system

       SBrkr	Standard Circuit Breakers & Romex
       FuseA	Fuse Box over 60 AMP and all Romex wiring (Average)	
       FuseF	60 AMP Fuse Box and mostly Romex wiring (Fair)
       FuseP	60 AMP Fuse Box and mostly knob & tube wiring (poor)
       Mix	Mixed
		
1stFlrSF: First Floor square feet
 
2ndFlrSF: Second floor square feet

LowQualFinSF: Low quality finished square feet (all floors)

GrLivArea: Above grade (ground) living area square feet

BsmtFullBath: Basement full bathrooms

BsmtHalfBath: Basement half bathrooms

FullBath: Full bathrooms above grade

HalfBath: Half baths above grade

Bedroom: Bedrooms above grade (does NOT include basement bedrooms)

Kitchen: Kitchens above grade

KitchenQual: Kitchen quality

       Ex	Excellent
       Gd	Good
       TA	Typical/Average
       Fa	Fair
       Po	Poor
       	
TotRmsAbvGrd: Total rooms above grade (does not include bathrooms)

Functional: Home functionality (Assume typical unless deductions are warranted)

       Typ	Typical Functionality
       Min1	Minor Deductions 1
       Min2	Minor Deductions 2
       Mod	Moderate Deductions
       Maj1	Major Deductions 1
       Maj2	Major Deductions 2
       Sev	Severely Damaged
       Sal	Salvage only
		
Fireplaces: Number of fireplaces

FireplaceQu: Fireplace quality

       Ex	Excellent - Exceptional Masonry Fireplace
       Gd	Good - Masonry Fireplace in main level
       TA	Average - Prefabricated Fireplace in main living area or Masonry Fireplace in basement
       Fa	Fair - Prefabricated Fireplace in basement
       Po	Poor - Ben Franklin Stove
       NA	No Fireplace
		
GarageType: Garage location
		
       2Types	More than one type of garage
       Attchd	Attached to home
       Basment	Basement Garage
       BuiltIn	Built-In (Garage part of house - typically has room above garage)
       CarPort	Car Port
       Detchd	Detached from home
       NA	No Garage
		
GarageYrBlt: Year garage was built
		
GarageFinish: Interior finish of the garage

       Fin	Finished
       RFn	Rough Finished	
       Unf	Unfinished
       NA	No Garage
		
GarageCars: Size of garage in car capacity

GarageArea: Size of garage in square feet

GarageQual: Garage quality

       Ex	Excellent
       Gd	Good
       TA	Typical/Average
       Fa	Fair
       Po	Poor
       NA	No Garage
		
GarageCond: Garage condition

       Ex	Excellent
       Gd	Good
       TA	Typical/Average
       Fa	Fair
       Po	Poor
       NA	No Garage
		
PavedDrive: Paved driveway

       Y	Paved 
       P	Partial Pavement
       N	Dirt/Gravel
		
WoodDeckSF: Wood deck area in square feet

OpenPorchSF: Open porch area in square feet

EnclosedPorch: Enclosed porch area in square feet

3SsnPorch: Three season porch area in square feet

ScreenPorch: Screen porch area in square feet

PoolArea: Pool area in square feet

PoolQC: Pool quality
		
       Ex	Excellent
       Gd	Good
       TA	Average/Typical
       Fa	Fair
       NA	No Pool
		
Fence: Fence quality
		
       GdPrv	Good Privacy
       MnPrv	Minimum Privacy
       GdWo	Good Wood
       MnWw	Minimum Wood/Wire
       NA	No Fence
	
MiscFeature: Miscellaneous feature not covered in other categories
		
       Elev	Elevator
       Gar2	2nd Garage (if not described in garage section)
       Othr	Other
       Shed	Shed (over 100 SF)
       TenC	Tennis Court
       NA	None
		
MiscVal: $Value of miscellaneous feature

MoSold: Month Sold (MM)

YrSold: Year Sold (YYYY)

SaleType: Type of sale
		
       WD 	Warranty Deed - Conventional
       CWD	Warranty Deed - Cash
       VWD	Warranty Deed - VA Loan
       New	Home just constructed and sold
       COD	Court Officer Deed/Estate
       Con	Contract 15% Down payment regular terms
       ConLw	Contract Low Down payment and low interest
       ConLI	Contract Low Interest
       ConLD	Contract Low Down
       Oth	Other
		
SaleCondition: Condition of sale

       Normal	Normal Sale
       Abnorml	Abnormal Sale -  trade, foreclosure, short sale
       AdjLand	Adjoining Land Purchase
       Alloca	Allocation - two linked properties with separate deeds, typically condo with a garage unit	
       Family	Sale between family members
       Partial	Home was not completed when last assessed (associated with New Homes)

In [126]:
df.columns

Index(['Id', 'MSSubClass', 'MSZoning', 'LotArea', 'Street', 'LotShape',
       'LandContour', 'Utilities', 'LotConfig', 'LandSlope', 'Neighborhood',
       'Condition1', 'Condition2', 'BldgType', 'HouseStyle', 'OverallQual',
       'OverallCond', 'YearBuilt', 'YearRemodAdd', 'RoofStyle', 'RoofMatl',
       'Exterior1st', 'Exterior2nd', 'ExterQual', 'ExterCond', 'Foundation',
       'BsmtFinSF1', 'BsmtFinSF2', 'BsmtUnfSF', 'TotalBsmtSF', 'Heating',
       'HeatingQC', 'CentralAir', 'Electrical', '1stFlrSF', '2ndFlrSF',
       'LowQualFinSF', 'GrLivArea', 'BsmtFullBath', 'BsmtHalfBath', 'FullBath',
       'HalfBath', 'BedroomAbvGr', 'KitchenAbvGr', 'KitchenQual',
       'TotRmsAbvGrd', 'Functional', 'Fireplaces', 'GarageType', 'GarageYrBlt',
       'GarageFinish', 'GarageCars', 'GarageArea', 'GarageQual', 'GarageCond',
       'PavedDrive', 'WoodDeckSF', 'OpenPorchSF', 'EnclosedPorch', '3SsnPorch',
       'ScreenPorch', 'PoolArea', 'MiscVal', 'MoSold', 'YrSold', 'SaleType',
       'SaleCo